### PR TITLE
Add REST endpoint for creating accounts

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

It's required to create accounts from the REST API

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201095311341924/1202790039552157/f) _(internal)_

## 🔍 What does this change?

- Add `AccountStore`, where you may insert accounts
- Add REST endpoint, which generates an ID and inserts the account into the database
- Adjust the REST endpoint tests to use the REST API to create an account (instead of using the root account)
- Adjust the integration tests to use the trait instead of the free function (which is now removed)

## 📜 Does this require a change to the docs?

The OpenAPI specs were adjusted

## 🛡 What tests cover this?

- REST API tests
- Integration tests

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/api/rest/account.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?;
+
+    let account_id = AccountId::new(Uuid::new_v4());
+    store
+        .insert_account_id(account_id)
+        .await
+        .map_err(|report| {
+            // TODO: consider adding the data type, or at least its URI in the trace
+            tracing::error!(error=?report, "Could not create data type");
```
Comment (benwerner01) [inline packages/graph/hash_graph/lib/graph/src/api/rest/account.rs:64]:
```suggestion
            tracing::error!(error=?report, "Could not create account id");
```

**Review by benwerner01 — COMMENTED** [review]
Looks good to me, thank you Tim! Spotted one small issue with an error message

File: packages/graph/hash_graph/lib/graph/src/api/rest/account.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        )
+    }
+}
+
+#[utoipa::path(
+    post,
+    path = "/accounts",
+    tag = "Account",
+    responses(
+        (status = 201, content_type = "application/json", description = "The schema of the created data type", body = AccountId),
```
Comment (TimDiekmann) [inline packages/graph/hash_graph/lib/graph/src/api/rest/account.rs:45]:
```suggestion
        (status = 201, content_type = "application/json", description = "The schema of the created account", body = AccountId),
```

File: packages/graph/hash_graph/lib/graph/src/api/rest/account.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/api/rest/account.rs]: Good spot, also found a similar error. Thanks!

**Review by benwerner01 — APPROVED** [review]
Thanks Tim, this looks good to me and is working for me locally 👍

Perhaps it's worth waiting for a review of the rust code from @Alfred-Mountfield or @thehabbos007 before merging.

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**